app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -241,6 +256,33 @@ private void handleDataReceived(RightMeshEvent e) {
         } catch (InvalidProtocolBufferException ignored) { /* Ignore malformed messages. */ }
     }
 
+    /**
+     * Handles a {@link KeyExchangeResultEvent} from RightMesh. Stores whether an online user can
+     * support encrypted messages if successful, or re-attempts the exchange if unsuccessful.
+     *
+     * @param e event passed from RightMesh
+     */
+    private void handleKeyExchangedResult(RightMeshEvent e) {
+        KeyExchangeResultEvent event = (KeyExchangeResultEvent) e;
+        if (event.status == MeshManager.SUCCESS) {
+            User user = users.get(event.peerUuid);
+            if (user != null) {
+                user.setIsEncrypted(true);
+                users.put(event.peerUuid, user);
+                updateInterface();
+            }
+        } else if (event.status == MeshManager.FAILURE) {
+            // Key exchange has timed out, try again.
+            try {
+                meshManager.exchangeKeys(event.peerUuid, MESH_PORT, 2000);
```

Comment:
Whats the purpose of port 2000. Should this be defined somewhere as TEST_PORT or something of that nature?

Reply:
`2000` is a timeout time in milliseconds.

Reply:
Could still store it somewhere I guess. :blush: